libs/@local/hash-graphql-shared/src/graphql/type-defs/knowledge/entity.typedef.ts

```diff
@@ -182,6 +229,29 @@ export const entityTypedef = gql`
       The ids of the possible entity types that inferred entities may be of.
       """
       entityTypeIds: [VersionedUrl!]!
+      """
+      The model to use for inference. Defaults to "gpt-4-0613".
+      """
+      model: String
+      """
+      The maximum amount of tokens to generate. If not specified, no limit is applied.
+      """
+      maxTokens: Int
+      """
+      Whether to allow empty results. Defaults to true.
+      """
+      allowEmptyResults: Boolean
+      """
+      The validation to apply to the inferred entities. Defaults to full. Possible values are:
+      - full: All entities must be valid.
+      - partial: Entity properties has to be valid but the "required" condition is omitted
+      - none: No validation is applied.
+      """
+      validation: EntityValidation
+      """
+      The temperature to use for inference. Defaults to 0.0.
+      """
+      temperature: Float
```

Comment:
We can add the defaults to the schema, and by doing so the GraphQL server will populate the default values if the caller doesn't provide any. One advantage of this is that we can change the defaults of the workflow without changing the defaults here (since the resolver will actually passing explicit values all the time for these properties, rather than nothing)

```suggestion
      model: String = "gpt-4-0613"
      """
      The maximum amount of tokens to generate. If not specified, no limit is applied.
      """
      maxTokens: Int
      """
      Whether to allow empty results. Defaults to true.
      """
      allowEmptyResults: Boolean = true
      """
      The validation to apply to the inferred entities. Defaults to full. Possible values are:
      - full: All entities must be valid.
      - partial: Entity properties has to be valid but the "required" condition is omitted
      - none: No validation is applied.
      """
      validation: EntityValidation = FULL
      """
      The temperature to use for inference. Defaults to 0.0.
      """
      temperature: Float = 0.0
```

Reply:
I would probably don't set any defaults in the workflow then, what do you think?

Reply:
Sounds good

Reply:
Oh, the TypeScript types for the resolver will not understand that these values will be defined, which might be a problem?

Reply:
I made all fields required in bdebdd6b848a8221c7fafc3c12cd58dd898a572d